### ISSUE TITLE
Compiling Godot 2 fails with clang 6 (-Wc++11-narrowing errors)

### ISSUE THREAD

**Issue by LinuxUserGD** [godotengine/godot#17725]
**Godot version:**
2.1 branch

**OS/device including version:**
Antergos OS (Linux)

**Issue description:**
Tried to compile Godot 2 from source.

**Steps to reproduce:**
Command: ```scons p=x11 bits=64 use_llvm=yes -j 8```
Error Output:
```Swift
[Initial build] Compiling ==> modules/chibi/register_types.cpp
platform/x11/os_x11.cpp:1401:37: error: non-constant-expression cannot be
      narrowed from type 'unsigned long' to 'int' in initializer list
      [-Wc++11-narrowing]
        Property p = { ret, actual_format, nitems, actual_type };
                                           ^~~~~~
platform/x11/os_x11.cpp:1401:37: note: insert an explicit cast to silence this
      issue
        Property p = { ret, actual_format, nitems, actual_type };
                                           ^~~~~~
                                           static_cast<int>( )
Ranlib Library         ==> main/tests/libtests.x11.tools.64.llvm.a
[Initial build] Compiling ==> modules/cscript/godot_c.cpp
[Initial build] Compiling ==> modules/cscript/register_types.cpp
[Initial build] Compiling ==> modules/dds/register_types.cpp
[Initial build] Compiling ==> modules/dds/texture_loader_dds.cpp
[Initial build] Compiling ==> thirdparty/rg-etc1/rg_etc1.cpp
[Initial build] Compiling ==> modules/etc1/image_etc.cpp
[Initial build] Compiling ==> modules/etc1/register_types.cpp
1 error generated.
[Initial build] Compiling ==> modules/etc1/texture_loader_pkm.cpp
scons: *** [platform/x11/os_x11.x11.tools.64.llvm.o] Error 1
scons: building terminated because of errors.

```

**Comment by akien-mga** [comment]
What clang version are you using?

**Comment by akien-mga** [comment]
Fixed by 60835ce88.

**Comment by LinuxUserGD** [comment]
@akien-mga Can you please reopen the issue?

**Comment by akien-mga** [comment]
It would have been better to open a new issue since it's about different errors (the initial bug report is fixed), but well, now that it's reopened let's use it.

**Comment by akien-mga** [comment]
Fixed by b29b35cee.
